### PR TITLE
Added support for desktop notifications

### PR DESCRIPTION
**Pull request by cajus**
So lets discuss how to add tests for that ;-)

### CODE REVIEW

File: framework/source/class/qx/bom/Notification.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+ *
+ * // Enable button if supported
+ * button.setEnabled(qx.core.Environment.get("html.notification"));
+ *
+ * </pre>
+ *
+ * **Note**
+ *
+ * A notification can only be send if the user allows these notifications to
+ * be shown. <code>qx.bom.Notification</code> automatically tries to
```
Comment (oetiker) [inline framework/source/class/qx/bom/Notification.js]: sen**t**

File: framework/source/class/qx/bom/Notification.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+ * trigger a browser dialog which asks the user for permission.
+ *
+ * But there is a restriction: the dialog will only show up,
+ * if the initial request was triggered from within an
+ * interactive event handler. So there is most likely a mouse or key
+ * event involved initially to trigger it properly.
+ *
+ * For real life applications that means that you may add a <code>request()</code>
+ * call i.e. to your applications login button to let the browser ask for
```
Comment (oetiker) [inline framework/source/class/qx/bom/Notification.js]: this means that

File: framework/source/class/qx/bom/Notification.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+ *
+ * A notification can only be send if the user allows these notifications to
+ * be shown. <code>qx.bom.Notification</code> automatically tries to
+ * trigger a browser dialog which asks the user for permission.
+ *
+ * But there is a restriction: the dialog will only show up,
+ * if the initial request was triggered from within an
+ * interactive event handler. So there is most likely a mouse or key
+ * event involved initially to trigger it properly.
+ *
```
Comment (oetiker) [inline framework/source/class/qx/bom/Notification.js]: maybe: 

But there is a restriction: the dialog will only show up if it is triggered by code that is running inside a request handler for an interactive browser event like a mouse click or a keyboard interaction.

**Comment by qx-pages-deployer** [comment]
My english is a bit rusty ;-) I'll update these. I'd like to skip multiple builds for travis, so just tell me if that's all and I'll update the PR.

**Comment by oetiker** [comment]
that is all that I noticed for the moment :)

**Comment by level420** [comment]
I support this PR.
Merging.

**Pull request merged**